substrate/frame/nomination-pools/src/adapter.rs

```diff
@@ -262,7 +262,8 @@ impl<T: Config, Staking: StakingInterface<Balance = BalanceOf<T>, AccountId = T:
 		pool_account: Pool<Self::AccountId>,
 		_: Member<Self::AccountId>,
 	) -> BalanceOf<T> {
-		T::Currency::balance(&pool_account.0).saturating_sub(Self::active_stake(pool_account))
+		// free/liquid balance of the pool account.
+		T::Currency::balance(&pool_account.0)
```

Comment:
This is the doc for `Currency::balance` is:
```
	/// Get the balance of `who` which does not include funds which are exclusively allocated to
	/// subsystems of the chain ("on hold" or "reserved").
	///
	/// In general this isn't especially useful outside of tests, and for practical purposes, you'll
	/// want to use [`Inspect::reducible_balance`].
```

So balance include the `frozen` amount. Later we do some transfer with `Currency::transfer` which implies `Polite` fortitude and thus can't transfer the `frozen` amount.

Maybe we want to use `reducible_balance(account, Expandable, Polie)` here? or do I miss something about slashing frozen amount?

Reply:
With current locks, its possible to stake all your funds including ED. I tried to retain that behaviour by adding provider to staking account. Therefore, you could stake all your free balance including ED, and so I used `Currency::balance`.

Update though: after some discussion with the staking team, we want stakers to maintain ED and I will make necessary updates to the PR.

Reply:
I meant about the `frozen` amount. calling `balance` here doesn't return the liquid balance. it returns the balance except holds, including frozen and ed.

I guess it works because the `pool_account` will never have `frozen` amounts.